test/functional/connection.test.js

```diff
@@ -517,7 +517,7 @@ describe('Connection', function() {
   });
 
   it('should be able to connect again after close', function() {
-    return withClient(this.configuration.newClient(), client => done => {
+    return withClient.call(this, (client, done) => {
```

Comment:
this should be equivalent to just calling `withClient` within the scope, why was this change needed?

Reply:
Without it I get `TypeError: Cannot read property 'configuration' of undefined` referencing [this line](https://github.com/mongodb/node-mongodb-native/pull/2355/files#diff-fd8c048cd2959e76db634d7719c6c1e3R92).